src/controller/webrtc/WebRTC.cpp

```diff
@@ -0,0 +1,270 @@
+
+    // Create a new peer connection
+    client->pc = std::make_shared<rtc::PeerConnection>(client->config);
+
+    // Set up event handlers for the peer connection
+    client->pc->onLocalDescription([client](rtc::Description description) {
+        std::string desc_string = std::string(description);
+        ChipLogProgress(NotSpecified, "Local Description (Paste this to the other peer): %s", desc_string.c_str());
+        if (client->offerCb)
+        {
+            client->offerCb(desc_string.c_str(), client->client_id);
+        }
+    });
+
+    client->pc->onLocalCandidate([client](rtc::Candidate candidate) {
+        std::string cand_string = std::string(candidate);
+        ChipLogProgress(NotSpecified, "Local Candidate (Paste this to the other peer after the local description): %s",
+                        cand_string.c_str());
+        if (client->iceCb)
+        {
+            client->iceCb(cand_string.c_str(), client->client_id);
+        }
+    });
+
+    client->pc->onStateChange([client](rtc::PeerConnection::State state) {
+        ChipLogProgress(NotSpecified, "[State: %u]", static_cast<unsigned>(state));
+        if (state == rtc::PeerConnection::State::Connected)
+        {
+            if (client->peerConnectedCb)
+            {
+                client->peerConnectedCb(client->client_id);
+            }
+        }
+        else if (state == rtc::PeerConnection::State::Disconnected)
+        {
+            if (client->peerDisconnectedCb)
+            {
+                client->peerDisconnectedCb(client->client_id);
+            }
+        }
+    });
+
+    client->pc->onGatheringStateChange([](rtc::PeerConnection::GatheringState state) {
+        ChipLogProgress(NotSpecified, "[Gathering State: %u]", static_cast<unsigned>(state));
+    });
+
+    // Create a data channel for communication
+    auto dc = client->pc->createDataChannel("test");
+
+    // Set up event handlers for the data channel
+    dc->onOpen([&]() { ChipLogProgress(NotSpecified, "[DataChannel open: %s]", dc->label().c_str()); });
+    dc->onClosed([&]() { ChipLogProgress(NotSpecified, "[DataChannel closed: %s]", dc->label().c_str()); });
+    dc->onMessage([](auto data) {
+        if (std::holds_alternative<std::string>(data))
+        {
+            ChipLogProgress(NotSpecified, "[Received message: %s]", std::get<std::string>(data).c_str());
+        }
+    });
+}
+
+// Function to destroy a WebRTC client
+void DestroyClient(void * Client) {}
+
+// API to close the peer connection and free up resources
+void ClosePeerConnection(void * Client)
+{
+    if (Client == nullptr)
+    {
+        ChipLogError(NotSpecified, "Client is null");
+        return;
+    }
+
+    WebRTCClient * rtcClient = static_cast<WebRTCClient *>(Client);
+    if (rtcClient->pc == nullptr)
+    {
+        ChipLogError(NotSpecified, "PeerConnection is null");
+        return;
+    }
+
+    rtcClient->pc->close();
+    delete rtcClient;
```

Comment:
Having dedicated destroy function which does nothing and a close function which actually destroys the object is kinda misleading...